pkg/agent/run.go

```diff
@@ -84,6 +84,12 @@ func getConfiguration(ctx context.Context) (Config, *client.PreflightClient) {
 		}
 	}
 
+	// if period is set in the config, then use that
```

Comment:
Looks good:+1: But could we do the assignment to Period outside `getConfiguration`?
Does the config has priority over the cli args?

Reply:
I moved it to the main loop in fe707383d090dbe264fcafd4d2d541e790bd8e27. I think this is what we want. Other option is to set the default for the period to 0 and fail if it's not set, then they set it in the flag, or in config, where the flag takes priority.

Actually, I like that better.

Reply:
18e1ebcb3ef8a766a57637fb70d1a637bdc5f293